src/common/snippets/src/op/subgraph.cpp

```diff
@@ -377,7 +387,14 @@ void Subgraph::data_flow_transformations(const BlockedShapeVector& blocked_input
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::op::data_flow_transformations")
 
     ov::snippets::pass::Manager manager;
-    if (!blocked_input_shapes.empty())
+    bool is_gn_subgraph = false;
+    for (const auto& op : body_ptr()->get_ordered_ops()) {
+        if (ov::is_type<ov::op::v12::GroupNormalization>(op)) {
+            is_gn_subgraph = true;
+            break;
+        }
+    }
```

Comment:
```suggestion
const auto& ops = body_ptr()->get_ordered_ops();
const auto is_gn_subgraph = std::any_of(ops.cbegin(), ops.cend(), [](const NodePtr& op) { return ov::is_type<ov::op::v12::GroupNormalization>(op); });
```

Reply:
Also could you please add the comment why we disable canonicalization in GN case?

@IvanNovoselov FYI. We disable canonicalization if there is GN as we discussed.

Reply:
Moreover, we can use here `PassConfig` to disable some transformations instead of using `if-else`. Probably not need to do in this PR - it's just some thoughts to do it in the near future as refactoring 😃

Reply:
Updated and code explanation comments is added. I have tried pass_config, seems like there is some problem. Let's refactor and fix it in next PR. Thanks Alexandra!